pilot/pkg/serviceregistry/kube/controller/ambient/waypoints.go

```diff
@@ -146,7 +146,9 @@ func fetchWaypointForWorkload(ctx krt.HandlerContext, Waypoints krt.Collection[W
 // defaultNamespace avoids the need to infer when object meta from a namespace was given
 func getUseWaypoint(meta metav1.ObjectMeta, defaultNamespace string) (named *krt.Named, isNone bool) {
 	if labelValue, ok := meta.Labels[constants.AmbientUseWaypointLabel]; ok {
-		if labelValue == "#none" || labelValue == "~" {
+		// NOTE: this means Istio reserves the word "none" in this field with a special meaning
+		//   a waypoint named "none" cannot be used and will be ignored
+		if labelValue == "none" {
```

Comment:
How can we make sure `none` is not a valid waypoint name

Reply:
https://github.com/istio/istio/pull/50842 was created to implements preventing istioctl users from choosing the reserved word `none` for their waypoint's name